src/goto-analyzer/static_verifier.cpp

```diff
@@ -27,6 +27,39 @@ struct static_verifier_resultt
   irep_idt function_id;
 };
 
+void static_verifier(
+  const abstract_goto_modelt &abstract_goto_model,
+  const ai_baset &ai,
+  propertiest &properties)
+{
+  auto const ns = namespacet{abstract_goto_model.get_symbol_table()};
```

Comment:
I have complained about this before and I'll keep doing so: why is the above better than `const namespacet ns{abstract_goto_model.get_symbol_table()};`? To me it seems that would be shorter and avoid any doubt about 1) the type of the object and 2) whether some `operator=` is being invoked or not. And if "consistency" is part of the rationale: in fact the above is the only case of `auto` that's not in violation of our current coding guidelines...

I keep going after this, because I'm worried that we will at some point arrive at write-only code. Explicitly spelling out types may seem like a nuisance to the person writing the code, but they are really helpful to any reader. In the code below, I actually have no idea what the types of `property_status` or `predecessor` are. Alas I'm actually unable to judge whether (not) using references is the right thing.

Reply:
I have to say I share the concerns of @tautschnig too, though he's put them more eloquently than I probably would have done. I'm not a huge fan of `auto` in anything other than very trivial contexts.

Reply:
I don't omit the types here because I don't care about readability, but rather because in my opinion they're not very useful for readability; I did another scan over the code and tried to clarify as much a possible, adding types where I thought they're useful. I kept the `auto` in the for loop and for `property_status` - the former because it'd otherwise just be `propertiest::value_type` which IMHO doesn't really add much useful information, and the latter because the type would be `property_statust` which again doesn't really communicate anything over the variable name.